c/Makefile

```diff
@@ -51,18 +53,45 @@ CFLAGS += -std=c99 -O
 # ---- Targets to build ----
 
 LIB = qrcodegen
-LIBFILE = lib$(LIB).a
+ARFILE = lib$(LIB).a
+LIBFILE = lib$(LIB).so
+# Bump the soname number when the ABI changes and gets incompatible
+SO_NAME = $(LIBFILE).1
+REAL_NAME = $(LIBFILE).$(VERSION)
+HEADERS = qrcodegen.h
 LIBOBJ = qrcodegen.o
 MAINS = qrcodegen-demo qrcodegen-test qrcodegen-worker
 
+# define paths to install
+INCLUDEDIR ?= $(DESTDIR)/usr/include/qrcodegen
+LIBDIR ?= $(DESTDIR)/usr/lib
```

Comment:
It would be nice if we can have these support a `PREFIX` variable since some systems like *BSD prefer installing 3rd packages to `/usr/local` but not `/usr`

Reply:
(there are also systems without /usr at all, e.g. NixOS and GuixSD)

Reply:
Added prefix, the variables can be overriden so the matter is just default.